torch/distributed/checkpoint/filesystem.py

```diff
@@ -585,7 +585,7 @@ def finish(self, metadata: Metadata, results: List[List[WriteResult]]) -> None:
             if self.sync_files:
                 try:
                     os.fsync(metadata_file.fileno())
-                except AttributeError:
+                except (AttributeError, UnsupportedOperation):
                     os.sync()
```

Comment:
We may want to debug log these errors?

Reply:
I thought about that, but I'm just following what has been done here (same issue, different place): https://github.com/pytorch/pytorch/pull/119287